core/src/banking_stage.rs

```diff
@@ -502,18 +502,19 @@ impl BankingStage {
             .iter()
             .filter_map(|p| {
                 if !p.meta.forwarded() && data_budget.take(p.meta.size) {
-                    Some((&p.data[..p.meta.size], tpu_forwards))
+                    Some(&p.data[..p.meta.size])
                 } else {
                     None
                 }
             })
             .collect();
 
+        // Based on conversations with sakridge, it's been decided to allow
+        // this number of successfully sent packets to be incorrect for now
         if !packet_vec.is_empty() {
             inc_new_counter_info!("banking_stage-forwarded_packets", packet_vec.len());
-            if let Err(SendPktsError::IoError(ioerr, num_failed)) = batch_send(socket, &packet_vec)
-            {
-                return (Err(ioerr), packet_vec.len().saturating_sub(num_failed));
+            if let Err(err) = batch_send_single_target(&packet_vec, tpu_forwards) {
+                return (Err(err), 0);
```

Comment:
maybe useful to have a counter here for when it does error?

Reply:
I'm a bit worried about performance in this case as well, we should at least add a counter for how long we are spending here. Also would be nice to test to see if we need to offload this to another thread or pool potentially.

Reply:
Agreed.  Every call site needs context-specific timing metrics

Reply:
Added the requested counters.